sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/AesKw.cs

```diff
@@ -0,0 +1,422 @@
+            }
+
+            public int OutputBlockSize
+            {
+                get { throw new NotImplementedException(); }
+            }
+
+            public int TransformBlock(byte[] inputBuffer, int inputOffset, int inputCount, byte[] outputBuffer, int outputOffset)
+            {
+                // No support for block-by-block transformation
+                throw new NotImplementedException();
+            }
+
+            public byte[] TransformFinalBlock(byte[] inputBuffer, int inputOffset, int inputCount)
+            {
+                if (inputBuffer == null)
+                    throw new ArgumentNullException(nameof(inputBuffer));
+
+                if (inputBuffer.Length == 0)
+                    throw new ArgumentNullException(nameof(inputBuffer), "The length of the inputBuffer parameter cannot be zero");
+
+                if (inputCount <= 0)
+                    throw new ArgumentOutOfRangeException(nameof(inputCount), "The inputCount parameter must not be zero or negative");
+
+                if (inputCount % 8 != 0)
+                    throw new ArgumentOutOfRangeException(nameof(inputCount), "The inputCount parameter must be a multiple of 64 bits");
+
+                if (inputOffset < 0)
+                    throw new ArgumentOutOfRangeException(nameof(inputOffset), "The inputOffset parameter must not be negative");
+
+                if (inputOffset + inputCount > inputBuffer.Length)
+                    throw new ArgumentOutOfRangeException(nameof(inputCount), "The sum of inputCount and inputOffset parameters must not be larger than the length of inputBuffer");
+
+                /*
+                    1) Initialize variables.
+
+                        Set A = C[0]
+                        For i = 1 to n
+                            R[i] = C[i]
+
+                    2) Compute intermediate values.
+
+                        For j = 5 to 0
+                            For i = n to 1
+                                B = AES-1(K, (A ^ t) | R[i]) where t = n*j+i
+                                A = MSB(64, B)
+                                R[i] = LSB(64, B)
+
+                    3) Output results.
+
+                    If A is an appropriate initial value (see 2.2.3),
+                    Then
+                        For i = 1 to n
+                            P[i] = R[i]
+                    Else
+                        Return an error
+                */
+
+                // A = C[0]
+                byte[] a = new byte[BlockSizeInBytes];
+
+                Array.Copy(inputBuffer, inputOffset, a, 0, BlockSizeInBytes);
+
+                // The number of input blocks
+                var n = (inputCount - BlockSizeInBytes) >> 3;
+
+                // The set of input blocks
+                byte[] r = new byte[n << 3];
+
+                Array.Copy(inputBuffer, inputOffset + BlockSizeInBytes, r, 0, inputCount - BlockSizeInBytes);
+
+                var encryptor = _aes.CreateDecryptor();
```

Comment:
"decryptor"

Reply:
This code is ported from directly track 1 and I'm leaving all the code in this class as is, since it's already been security reviewed for it's crypto implementation.

Reply:
Nit: sure, but this was just a poorly named local variable.